Add option to remove local repos that have been archived, add description for tasks, add typing and update readme

sopelj opened the pull request.

Commit:
Add option to remove local repos that are archived, add description for
tasks, add typing and update readme

Commit:
Add exclude option for removing archived repos

Commit:
Add pre-commit and remaining typing

fblackburn1 commented:
I would be nice to merge this PR. Is it ready or still in progress?
If ready, don't forget to add `mergeit` + assigne a reviewer :smiley:

sopelj commented:
> I would be nice to merge this PR. Is it ready or still in progress? If ready, don't forget to add `mergeit` + assigne a reviewer smiley

I think it could be handy too. It was ready last sprint, I just added some extra things yesterday. I added the tag, but I never know who to get to review things.

sduthil committed:
typos

sduthil commented:
Thanks a lot Jesse!
